iothub_client/src/iothub_client_core.c

```diff
@@ -1691,13 +1699,21 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_SetOption(IOTHUB_CLIENT_CORE_HANDLE iotHub
         }
         else
         {
-            /*Codes_SRS_IOTHUBCLIENT_02_038: [If optionName doesn't match one of the options handled by this module then IoTHubClient_SetOption shall call IoTHubClientCore_LL_SetOption passing the same parameters and return what IoTHubClientCore_LL_SetOption returns.] */
-            result = IoTHubClientCore_LL_SetOption(iotHubClientInstance->IoTHubClientLLHandle, optionName, value);
-            if (result != IOTHUB_CLIENT_OK)
+            /* Codes_SRS_IOTHUBCLIENT_41_001 [ If parameter `optionName` is `OPTION_CONVENIENCE_LOOP_TIME` then `IoTHubClient_SetOption` shall set `loop_timeout` parameter of `IoTHubClientInstance` ]*/ 
+            if (strcmp(OPTION_CONVENIENCE_LOOP_TIME, optionName) == 0)
```

Comment:
>OPTION_CONVENIENCE_LOOP_TIME [](start = 23, length = 28)

Is there any timeout within our system that might conflict with this? For instance, can I set message timeout < loop time and never send a message? We should help avoid these scenarios. Please change the telemetry sample and try this out. This is why a design within the Github issue is preferred over design discussions within PRs :) 
For reference, please see the Current Configuration Options section of this article - [https://github.com/Azure/azure-iot-sdk-c/blob/master/doc/connection_and_messaging_reliability.md](https://github.com/Azure/azure-iot-sdk-c/blob/master/doc/connection_and_messaging_reliability.md)

Reply:
there was actually a bug in the message timeout (it would essentially never be called :) ) so I've also fixed that. I'll add validation for message timeout and for this to make sure the two options don't step on each other's toes.